File: feed-service/src/feed/dto/FeedBatchEvent.ts
import { ContentType } from 'src/enums/ContentType';
import { FeedSource } from 'src/enums/FeedSource';

export class FeedBatchEvent {
  followerIds!: string[];
  contentId!: number;
  contentType!: ContentType;
  likeCount!: number;
  commentCount!: number;
  saveCount!: number;
  timestamp!: string;
}

export class FeedBackfillEvent {
  followerId!: string;
  userId!: string;
}

export class FeedItemDto {
  contentId!: number;
  contentType!: ContentType;
  source!: FeedSource;
  score!: number;
  createdAt!: Date;
}

File: follow-service/src/main/java/com/example/follow_service/events/FeedBatchEvent.java
package com.example.follow_service.events;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
public class FeedBatchEvent {
    private List<String> followerIds;
    private Long contentId;
    private String contentType;
    private Long likeCount;
    private Long saveCount;
    private Long commentCount;
    private String timestamp;
}
